### ISSUE TITLE
Feature request: extend scope of `"python.tensorBoard.logDirectory"` and/or support variable substitution

### ISSUE THREAD

**Issue by bersbersbers** [microsoft/vscode-jupyter#8802]
TLDR: `"python.tensorBoard.logDirectory"` is user-level only and does not seem to support variable substitution which, especially in combination, makes it very hard to use productively.

I would love to add my thoughts to https://github.com/microsoft/vscode-python/discussions/16073 or https://github.com/microsoft/vscode-python/issues/17222, but I missed them over the fall and now they are both closed.

My main argument is that not only should `"python.tensorBoard.logDirectory"` be a workspace setting, but also that it makes little sense as a user setting at all - should it remain a user setting, I would be surprised if many people successfully used it who are using more than one workspace. This is because the TensorBoard log dir is set
1. either relatively, so it depends on the current working directory,
2. or absolutely, so it has be fully defined set in the workspace code.

The problem with 1. is obvious - even the same code, run in two different workspaces, will produce two different directories. This could be taken care of by variable substition, such as `"python.tensorBoard.logDirectory": "${workspaceFolder}/results/tensorboard"`, but I seem to be unable to pull this off:
![image](https://user-images.githubusercontent.com/12128514/150870983-dd097bef-3134-4df3-af6e-d9e0e23e1f08.png)

It seems like "Visual Studio Code supports variable substitution in ... some select settings" does not include this setting.

The problem with 2 is that, if your workspace code (the writer) uses a fixed TensorBoard logdir location, you have no way of ensuring (in the workspace) that the effective VS Code setting (the reader) is in sync with that. I am not sure defining a workspace-level launch configuration (for the writer) pulling in a user-level logdir location setting is very popular..

**Comment by rchiodo** [comment]
@joyceerhl would you happen to remember if there was a reason why ```"python.tensorBoard.logDirectory"``` is user level instead of workspace?

**Comment by joyceerhl** [comment]
@bersbersbers linked the issue and discussion where we mentioned waiting for feedback before making it a workspace setting because we'd need to account for the multi-root workspace case.

**Comment by rchiodo** [comment]
Variable substitution should be easy to fix. This code here:
https://github.dev/microsoft/vscode-python/blob/e157943a1dc7edfff238e1acdf7e57e3dd6558fc/src/client/tensorBoard/tensorBoardSession.ts#L342

Just needs to use the config service instead of getting the value directly.

**Comment by PillarsZhang** [comment]
I also need these features: https://github.com/microsoft/vscode-jupyter/issues/9677
This is my attempt, including variable and relative path resolution: https://github.com/PillarsZhang/vscode-python/commit/182af46a33ecc5cc5453d8dcc5f0d62bc52899ac
I also hope to set logdir separately for each workspace, which is more flexible.

**Comment by evanphilip** [comment]
@DonJayamanne is there any roadblocks to implementing this feature? Especially the PR from @PillarsZhang https://github.com/PillarsZhang/vscode-python/commit/182af46a33ecc5cc5453d8dcc5f0d62bc52899ac

**Comment by evanphilip** [comment]
The fork from @PillarsZhang wouldn't merge easily, so I implemented the same fix on a new fork and created pull request [20103](https://github.com/microsoft/vscode-python/pull/20103) . 

Sorry about the spam here; my experience is limited to Git

**Comment by andreamah** [comment]
are there some verification steps that we can follow to make sure that this is resolved?
